web/turkey/survey/auditors.py

```diff
@@ -7,72 +7,405 @@
-    # 'focus_changes': '',  # timestamped
+    'clicks_total': 'AuditorClicksTotal',
+    'clicks_specific': 'AuditorClicksSpecific',
+    'focus_changes': 'AuditorFocusChanges',  # timestamped
+    # There are fields total tabs, total
+    # backspaces, and count of unique characters
+    # in the paper above, but these can
+    # be extracted from aggregations on
+    # keypresses_specific
+    'keypresses_total': 'AuditorKeypressesTotal',
+    'keypresses_specific': 'AuditorKeypressesSpecific',
+    'mouse_movement_total': 'AuditorMouseMovementTotal',
+    'mouse_movement_specific': 'AuditorMouseMovementSpecific',
+    'on_focus_time': 'AuditorOnFocusTime',
+    'pastes_total': 'AuditorPastesTotal',
+    'pastes_specific': 'AuditorPastesSpecific',
+    'recorded_time_disparity': 'AuditorRecordedTimeDisparity',
+    'scrolled_pixels_total': 'AuditorScrolledPixelsTotal', 
+    'scrolled_pixels_specific': 'AuditorScrolledPixelsSpecific',
+    'within_typing_delay': 'AuditorWithinTypingDelay',
+    'total_task_time': 'AuditorTotalTaskTime',
 }
 
+#before_typing_delay
+class AuditorBeforeTypingDelayData(AuditorData):
+    general_model = models.ForeignKey('AuditorBeforeTypingDelay')
+    milliseconds = models.IntegerField(
+        verbose_name=_('before typing delay'),
+        help_text=_('total time in milliseconds that the user took before typing'),
+        null=True  # can be null because user might never type
+        blank=True
+    )
+
+    class Meta(AuditorData.Meta):
+        abstract = False
+
+class AuditorBeforeTypingDelay(Auditor):
+    script_location = 'survey/js/auditors/before_typing_delay.js'
+    data_model = AuditorBeforeTypingDelayData
+
+    class Meta(Auditor.Meta):
+        abstract = False
+        verbose_name = _('Before Typing Delay Auditor')
+        verbose_name_plural = _('Before Typing Delay Auditors')
+
+
+#clicks_total
+class AuditorClicksTotalData(AuditorData):
+    general_model = models.ForeignKey('AuditorClicksTotal')
+    count = models.IntegerField(
+        verbose_name=_('clicks total'),
+        help_text=_('total number of times a user clicked')
+    )
+
+    class Meta(AuditorData.Meta):
+        abstract = False
+
+class AuditorClicksTotal(Auditor):
+    script_location = 'survey/js/auditors/clicks_total.js'
+    data_model = AuditorClicksTotalData
+
+    class Meta(Auditor.Meta):
+        abstract = False
+        verbose_name = _('Clicks Total Auditor')
+        verbose_name_plural = _('Clicks Total Auditors')
+
+
+#clicks_specific
+class AuditorClicksSpecificData(AuditorData):
+    general_model = models.ForeignKey('AuditorClicksSpecific')
+    _type = models.TextField(
+        verbose_name=_('clicks specific type')
+        help_text=_('DOM type of element that was clicked')
+    )
+    _id = models.TextField(
+        verbose_name=_('clicks specific id')
+        help_text=_('DOM ID of element that was clicked')
+        null=True
+        blank=True
+    )
+    _class = models.TextField(
```

Comment:
This is a little confusing to learn for someone who knows Python. Python naming convention is that weakly "private" members of a class should begin with an underscore. This is common enough that the behavior is specified in [PEP 0008](https://www.python.org/dev/peps/pep-0008/#naming-conventions) and in fact in some instances naming variables with a leading underscored affects the way the file is parsed. So this reads like you don't want people touching that part of the model. I also wouldn't be surprised if there's something deep in the Django code that treats variables with a leading underscore differently.

Could we just use "dom_id" or "dom_class"?